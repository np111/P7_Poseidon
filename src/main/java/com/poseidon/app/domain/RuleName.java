package com.poseidon.app.domain;

import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "rulename")
public class RuleName {
    // TODO: Map columns in data table RULENAME with corresponding java fields
}
