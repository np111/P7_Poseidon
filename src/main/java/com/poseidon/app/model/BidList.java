package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class BidList {
    private Long id;
    private String account;
    private String type;
    private Double bidQuantity;
    private Double askQuantity;
    private Double bid;
    private Double ask;
    private String benchmark;
    private ZonedDateTime bidListDate;
    private String commentary;
    private String security;
    private String status;
    private String trader;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
