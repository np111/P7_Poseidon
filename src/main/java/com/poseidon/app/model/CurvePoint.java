package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class CurvePoint {
    private Long id;
    private Long curveId;
    private ZonedDateTime asOfDate;
    private Double term;
    private Double value;
    private ZonedDateTime creationDate;
}
