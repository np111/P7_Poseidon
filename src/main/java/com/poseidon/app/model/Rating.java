package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Rating {
    private Long id;

    @NotBlank(message = "MoodysRating is mandatory")
    private String moodysRating;

    @NotBlank(message = "SandPRating is mandatory")
    private String sandPRating;

    @NotBlank(message = "FitchRating is mandatory")
    private String fitchRating;

    @NotNull(message = "Order is mandatory")
    private Byte orderNumber;
}
