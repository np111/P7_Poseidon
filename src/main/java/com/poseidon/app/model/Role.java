package com.poseidon.app.model;

public enum Role {
    USER,
    ADMIN,
}
