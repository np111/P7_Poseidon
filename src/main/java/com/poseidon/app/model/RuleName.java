package com.poseidon.app.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class RuleName {
    private Long id;
    private String name;
    private String description;
    private String json;
    private String template;
    private String sqlStr;
    private String sqlPart;
}
