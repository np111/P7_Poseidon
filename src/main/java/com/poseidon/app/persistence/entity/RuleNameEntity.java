package com.poseidon.app.persistence.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "rulename")
@NoArgsConstructor
@Data
public class RuleNameEntity {
    public RuleNameEntity(String name, String description, String json, String template, String sqlStr, String sqlPart) {
        this.name = name;
        this.description = description;
        this.json = json;
        this.template = template;
        this.sqlStr = sqlStr;
        this.sqlPart = sqlPart;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "json")
    private String json;

    @Column(name = "template")
    private String template;

    @Column(name = "sqlStr")
    private String sqlStr;

    @Column(name = "sqlPart")
    private String sqlPart;
}
