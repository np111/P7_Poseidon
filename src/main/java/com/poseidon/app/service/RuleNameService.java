package com.poseidon.app.service;

import com.poseidon.app.model.RuleName;
import com.poseidon.app.persistence.entity.RuleNameEntity;
import com.poseidon.app.persistence.mapper.RuleNameMapper;
import com.poseidon.app.persistence.repository.RuleNameRepository;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor(onConstructor = @__(@Autowired))
@Service
@Scope("singleton")
public class RuleNameService extends CrudService<RuleName, RuleNameEntity, Long> {
    private final @Getter RuleNameRepository repository;
    private final @Getter RuleNameMapper mapper;

    @Override
    protected void setModelId(RuleName user, Long id) {
        user.setId(id);
    }
}
