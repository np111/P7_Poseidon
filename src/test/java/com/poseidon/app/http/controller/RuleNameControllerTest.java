package com.poseidon.app.http.controller;

import com.poseidon.app.model.RuleName;
import com.poseidon.app.persistence.entity.RuleNameEntity;
import com.poseidon.app.service.RuleNameService;
import lombok.Getter;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;

@WebMvcTest(RuleNameController.class)
class RuleNameControllerTest extends AbstractCrudControllerTest<RuleNameService, RuleName, RuleNameEntity, Long> {
    @MockBean
    private @Getter RuleNameService service;

    public RuleNameControllerTest() {
        super("/ruleName", RuleName.class, Long.class);
    }

    @Override
    protected RuleName getModelA() {
        return RuleName.builder()
                .id(1L)
                .name("name")
                .description("description")
                .json("json")
                .template("template")
                .sqlStr("sqlStr")
                .sqlPart("sqlPart")
                .build();
    }

    @Override
    protected RuleName getModelB() {
        return RuleName.builder()
                .id(583352L)
                .name("fQQMUyhdfB")
                .description("8uNlqQUCE6")
                .json("TKNmqCNgmc")
                .template("d4uLWSNQ93")
                .sqlStr("wLmFRdlkBJ")
                .sqlPart("WbKnU8R4Mq")
                .build();
    }
}
