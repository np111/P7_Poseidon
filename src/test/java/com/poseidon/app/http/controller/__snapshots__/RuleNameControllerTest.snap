com.poseidon.app.http.controller.RuleNameControllerTest.add=[
-
> GET /ruleName/add
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Rule</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/ruleName/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"name\" class=\"col-sm-2 control-label\">Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"name\" placeholder=\"Name\" class=\"col-4\" name=\"name\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"description\" class=\"col-sm-2 control-label\">Description</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"description\" placeholder=\"Description\" class=\"col-4\" name=\"description\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"json\" class=\"col-sm-2 control-label\">json</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"json\" placeholder=\"json\" class=\"col-4\" name=\"json\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"template\" class=\"col-sm-2 control-label\">template</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"template\" placeholder=\"template\" class=\"col-4\" name=\"template\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlStr\" class=\"col-sm-2 control-label\">sql</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlStr\" placeholder=\"sql\" class=\"col-4\" name=\"sqlStr\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlPart\" class=\"col-sm-2 control-label\">sqlPart</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlPart\" placeholder=\"sqlPart\" class=\"col-4\" name=\"sqlPart\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      "
"      "
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/ruleName/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add rule\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.RuleNameControllerTest.delete=[
-
> GET /ruleName/delete/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /ruleName/delete/583352
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /ruleName/list
""
]


com.poseidon.app.http.controller.RuleNameControllerTest.list=[
-
> GET /ruleName/list
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <div class=\"col-6\">"
"      <a href=\"/bidList/list\">Bid List</a>&nbsp;|&nbsp;"
"      <a href=\"/curvePoint/list\">Curve Points</a>&nbsp;|&nbsp;"
"      <a href=\"/rating/list\">Ratings</a>&nbsp;|&nbsp;"
"      <a href=\"/trade/list\">Trade</a>&nbsp;|&nbsp;"
"      <a href=\"/ruleName/list\">Rule</a>"
"    </div>"
"    <div class=\"col-6 text-right\">"
"      Logged in user: <b class=\"user\"> test-username </b>"
"      <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"        <input type=\"submit\" value=\"Logout\"/>"
"      </form>"
"    </div>"
"  </div>"
"  <div class=\"row\"><h2>Rule List</h2></div>"
"  <div class=\"row\">"
"    <a href=\"/ruleName/add\" class=\"btn btn-primary btn-sm\">Add New</a>"
"    <table class=\"table table-bordered\">"
"      <thead>"
"      <tr>"
"        <th>Id</th>"
"        <th>Name</th>"
"        <th>Description</th>"
"        <th>json</th>"
"        <th>template</th>"
"        <th>sql</th>"
"        <th>sqlPart</th>"
"        <th>Action</th>"
"      </tr>"
"      </thead>"
"      <tbody>"
"      <tr>"
"        <td>1</td>"
"        <td>name</td>"
"        <td>description</td>"
"        <td>json</td>"
"        <td>template</td>"
"        <td>sqlStr</td>"
"        <td>sqlPart</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/ruleName/update/1\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/ruleName/delete/1\">Delete</a>"
"        </td>"
"      </tr>"
"      <tr>"
"        <td>583352</td>"
"        <td>fQQMUyhdfB</td>"
"        <td>8uNlqQUCE6</td>"
"        <td>TKNmqCNgmc</td>"
"        <td>d4uLWSNQ93</td>"
"        <td>wLmFRdlkBJ</td>"
"        <td>WbKnU8R4Mq</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/ruleName/update/583352\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/ruleName/delete/583352\">Delete</a>"
"        </td>"
"      </tr>"
"      </tbody>"
"    </table>"
"  </div>"
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.RuleNameControllerTest.showUpdate=[
-
> GET /ruleName/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /ruleName/update/583352
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Update New Rule</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/ruleName/update/583352\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"name\" class=\"col-sm-2 control-label\">Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"name\" placeholder=\"Name\" class=\"col-4\" name=\"name\" value=\"fQQMUyhdfB\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"description\" class=\"col-sm-2 control-label\">Description</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"description\" placeholder=\"Description\" class=\"col-4\" name=\"description\" value=\"8uNlqQUCE6\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"json\" class=\"col-sm-2 control-label\">json</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"json\" placeholder=\"json\" class=\"col-4\" name=\"json\" value=\"TKNmqCNgmc\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"template\" class=\"col-sm-2 control-label\">template</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"template\" placeholder=\"template\" class=\"col-4\" name=\"template\" value=\"d4uLWSNQ93\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlStr\" class=\"col-sm-2 control-label\">sql</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlStr\" placeholder=\"sql\" class=\"col-4\" name=\"sqlStr\" value=\"wLmFRdlkBJ\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlPart\" class=\"col-sm-2 control-label\">sqlPart</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlPart\" placeholder=\"sqlPart\" class=\"col-4\" name=\"sqlPart\" value=\"WbKnU8R4Mq\">"
"          "
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/ruleName/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update rule\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.RuleNameControllerTest.update=[
-
> POST /ruleName/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Update New Rule</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/ruleName/update/1\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"name\" class=\"col-sm-2 control-label\">Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"name\" placeholder=\"Name\" class=\"col-4\" name=\"name\" value=\"\">"
"          <p class=\"text-danger\">Name is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"description\" class=\"col-sm-2 control-label\">Description</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"description\" placeholder=\"Description\" class=\"col-4\" name=\"description\" value=\"\">"
"          <p class=\"text-danger\">Description is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"json\" class=\"col-sm-2 control-label\">json</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"json\" placeholder=\"json\" class=\"col-4\" name=\"json\" value=\"\">"
"          <p class=\"text-danger\">json is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"template\" class=\"col-sm-2 control-label\">template</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"template\" placeholder=\"template\" class=\"col-4\" name=\"template\" value=\"\">"
"          <p class=\"text-danger\">template is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlStr\" class=\"col-sm-2 control-label\">sql</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlStr\" placeholder=\"sql\" class=\"col-4\" name=\"sqlStr\" value=\"\">"
"          <p class=\"text-danger\">sql is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlPart\" class=\"col-sm-2 control-label\">sqlPart</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlPart\" placeholder=\"sqlPart\" class=\"col-4\" name=\"sqlPart\" value=\"\">"
"          <p class=\"text-danger\">sqlPart is mandatory</p>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/ruleName/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update rule\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /ruleName/update/1
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /ruleName/list
""
]


com.poseidon.app.http.controller.RuleNameControllerTest.validate=[
-
> POST /ruleName/validate
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Rule</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/ruleName/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"name\" class=\"col-sm-2 control-label\">Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"name\" placeholder=\"Name\" class=\"col-4\" name=\"name\" value=\"\">"
"          <p class=\"text-danger\">Name is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"description\" class=\"col-sm-2 control-label\">Description</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"description\" placeholder=\"Description\" class=\"col-4\" name=\"description\" value=\"\">"
"          <p class=\"text-danger\">Description is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"json\" class=\"col-sm-2 control-label\">json</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"json\" placeholder=\"json\" class=\"col-4\" name=\"json\" value=\"\">"
"          <p class=\"text-danger\">json is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"template\" class=\"col-sm-2 control-label\">template</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"template\" placeholder=\"template\" class=\"col-4\" name=\"template\" value=\"\">"
"          <p class=\"text-danger\">template is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlStr\" class=\"col-sm-2 control-label\">sql</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlStr\" placeholder=\"sql\" class=\"col-4\" name=\"sqlStr\" value=\"\">"
"          <p class=\"text-danger\">sql is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sqlPart\" class=\"col-sm-2 control-label\">sqlPart</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sqlPart\" placeholder=\"sqlPart\" class=\"col-4\" name=\"sqlPart\" value=\"\">"
"          <p class=\"text-danger\">sqlPart is mandatory</p>"
"        </div>"
"      </div>"
"      "
"      "
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/ruleName/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add rule\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /ruleName/validate
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /ruleName/list
""
]